crates/hotshot/src/traits/election/static_committee_leader_two_views.rs

```diff
@@ -1,202 +1,159 @@
+            committee_members
+                .iter()
+                .map(|member| member.stake_table_entry.clone())
+                .filter(|entry| entry.stake() > U256::zero())
+                .collect();
+
+        // Index the stake table by public key
+        let indexed_stake_table: BTreeMap<
+            TYPES::SignatureKey,
+            <TYPES::SignatureKey as SignatureKey>::StakeTableEntry,
+        > = members
+            .iter()
+            .map(|entry| (TYPES::SignatureKey::public_key(entry), entry.clone()))
+            .collect();
+
         Self {
-            all_nodes_with_stake: nodes_with_stake.clone(),
-            committee_nodes_with_stake: nodes_with_stake,
-            committee_nodes_without_stake: nodes_without_stake,
-            fixed_leader_for_gpuvid,
-            _type_phantom: PhantomData,
+            eligible_leaders,
+            stake_table: members,
+            indexed_stake_table,
             committee_topic,
+            #[cfg(feature = "fixed-leader-election")]
+            fixed_leader_for_gpuvid,
         }
     }
-}
-
-impl<TYPES, PUBKEY: SignatureKey + 'static> Membership<TYPES>
-    for StaticCommitteeLeaderForTwoViews<TYPES, PUBKEY>
-where
-    TYPES: NodeType<SignatureKey = PUBKEY>,
-{
-    /// Clone the public key and corresponding stake table for current elected committee
-    fn committee_qc_stake_table(&self) -> Vec<PUBKEY::StakeTableEntry> {
-        self.committee_nodes_with_stake.clone()
-    }
 
-    /// Get the network topic for the committee
-    fn committee_topic(&self) -> Topic {
-        self.committee_topic.clone()
-    }
-
-    /// Index the vector of public keys with the current view number
-    fn leader(&self, view_number: TYPES::Time) -> PUBKEY {
-        // two connsecutive views will have same index starting with even number.
-        // eg 0->1, 2->3 ... 10->11 etc
-        let index =
-            usize::try_from((*view_number / 2) % self.all_nodes_with_stake.len() as u64).unwrap();
-        let res = self.all_nodes_with_stake[index].clone();
-        TYPES::SignatureKey::public_key(&res)
+    /// Get the stake table for the current view
+    fn get_stake_table(
+        &self,
+    ) -> Vec<<<TYPES as NodeType>::SignatureKey as SignatureKey>::StakeTableEntry> {
+        self.stake_table.clone()
     }
 
-    fn has_stake(&self, pub_key: &PUBKEY) -> bool {
-        let entry = pub_key.stake_table_entry(1u64);
-        self.committee_nodes_with_stake.contains(&entry)
+    /// Get all members of the committee for the current view
+    fn get_committee_members(
+        &self,
+        _view_number: <TYPES as NodeType>::Time,
+    ) -> std::collections::BTreeSet<<TYPES as NodeType>::SignatureKey> {
+        self.stake_table
+            .iter()
+            .map(TYPES::SignatureKey::public_key)
+            .collect()
     }
 
-    fn stake(
+    /// Get the stake table entry for a public key
+    fn get_stake(
```

Comment:
Same remark here as below about trivial getters, and everywhere else in this file.